src/Apps/cudaPlanesweepMAPTk.cpp

```diff
@@ -414,8 +414,9 @@ int main(int argc, char* argv[])
                 farZ = zs[(int)((zs.size()-1)*(1-outlierThres))];
                 std::cout<<"    distances from camera are "<<zs[0]<<"->"<<zs[zs.size()-1]<<" ("<< zs.size()<<" visible points)"<<std::endl;
                 std::cout<<"    distances ["<<outlierThres*100<<"%->"<<(1.0-outlierThres)*100<<"%] are "<<nearZ<<"->"<<farZ<<std::endl;
-                nearZ *= (1-safeness_margin_factor);
-                farZ *= (1+safeness_margin_factor);
+                const double safeness_margin = safeness_margin_factor * (farZ - nearZ);
+                nearZ -= safeness_margin;
+                farZ += safeness_margin;
```

Comment:
Do you mean that expressing the margin as a percentage of the robust span of points is better that expressing it as an absolute percentage?
Both seem fine to me.